src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/PInvokeMethodFixupNode.cs

```diff
@@ -72,16 +82,29 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             // Module fixup cell
             builder.EmitPointerReloc(factory.PInvokeModuleFixup(_moduleName));
 
+            builder.EmitInt(_exactMatchOnly ? 0 : (int)_charSet);
+
             return builder.ToObjectData();
         }
 
         protected internal override int ClassCode => -1592006940;
 
         protected internal override int CompareToImpl(SortableDependencyNode other, CompilerComparer comparer)
         {
-            var compare = string.Compare(_moduleName, ((PInvokeMethodFixupNode)other)._moduleName);
-            if (compare != 0)
-                return compare;
+            var exactMatchCompare = _exactMatchOnly.CompareTo(((PInvokeMethodFixupNode)other)._exactMatchOnly);
+            if (exactMatchCompare != 0)
+            {
```

Comment:
Nit: would be nice to stay consistent about grouping things together and the use of curly braces for single-liners.